pennylane/transforms/decompositions/two_qubit_unitary.py

```diff
@@ -170,18 +170,31 @@ def _su2su2_to_tensor_products(U):
     # C1 C2^dag = a1 a2* I
     C12 = math.dot(C1, math.conj(math.T(C2)))
 
-    if not math.allclose(a1 * math.conj(a2), C12[0, 0]):
-        a2 *= -1
+    if not math.is_abstract(C12):
+        if not math.allclose(a1 * math.conj(a2), C12[0, 0]):
+            a2 *= -1
+    else:
+        sign_is_correct = math.allclose(a1 * math.conj(a2), C12[0, 0])
+        sign = (-1) ** (sign_is_correct + 1)
```

Comment:
`sign_is_correct + 1` :astonished: Very neat trick! :clap: Worth leaving a comment here on the "promoting to an integer" business so that readers are informed.